### PR TITLE
Masq integration

### PR DESCRIPTION
**Pull request by sujeebant**

### CODE REVIEW

File: src/adapters/poi/poi_store.js [hunk 1]
```diff
@@ -23,7 +23,8 @@ export default class PoiStore extends Poi {
       storedData = await store.getAllPois()
     } catch(e) {
       Telemetry.add(Telemetry.FAVORITE_ERROR_LOAD_ALL)
-      Error.sendOnce('favorite_panel', 'getAll', 'error getting pois', e)
+      Error.sendOnce('poi_store', 'getAll', 'error getting pois', e)
```
Comment (Julien-laville) [inline src/adapters/poi/poi_store.js]: thx ;)

File: src/adapters/store.js [hunk 2]
```diff
@@ -27,8 +35,23 @@ function Store() {
   }
 }
 
+Store.prototype.getUserInfo = async function() {
+  try {
+    if (! (await checkRegistered())) {
```
Comment (Julien-laville) [inline src/adapters/store.js]: the condition can be reversed to avoid that error prone syntax

File: src/adapters/store.js [hunk 3]
```diff
@@ -27,8 +35,23 @@ function Store() {
...
+      throw new Error('Not registered')
+    }
+    return await abstractStore.getUserInfo()
+  } catch (e) {
+    Error.sendOnce('store', 'getAllPois', 'error getting pois', e)
```
Comment (Julien-laville) [inline src/adapters/store.js]: Copy pasta error 🍝

File: src/adapters/store.js [hunk 4]
```diff
@@ -27,8 +35,23 @@ function Store() {
...
+    throw e
+  }
+}
+
 Store.prototype.getAllPois = async function() {
   try {
+    if (! (await checkRegistered())) {
```
Comment (Julien-laville) [inline src/adapters/store.js]: you need an abstraction to avoid this copy pasta sauce 🍝

File: src/adapters/store.js [hunk 5]
```diff
@@ -38,6 +61,10 @@ Store.prototype.getAllPois = async function() {
 
 Store.prototype.getLastLocation = async function() {
   try {
+    if (! (await checkRegistered())) {
+      return await fallbackStore.get(`qmaps_v${version}_last_location`)
+    }
+    console.log('registered')
```
Comment (Julien-laville) [inline src/adapters/store.js]: linter here

File: src/adapters/store.js [hunk 6]
```diff
@@ -47,18 +74,17 @@ Store.prototype.getLastLocation = async function() {
 
 Store.prototype.setLastLocation = async function(loc) {
   try {
+    if (! (await checkRegistered())) {
+      return await fallbackStore.set(`qmaps_v${version}_last_location`, loc)
```
Comment (Julien-laville) [inline src/adapters/store.js]: we need a better logic between fallback and masq

File: src/libs/local_store.js [hunk 7]
```diff
@@ -19,15 +19,31 @@ LocalStore.prototype.getAllPois = function() {
...
+LocalStore.prototype.isRegistered = function() {
+  return Promise.resolve(true)
+}
+
 LocalStore.prototype.onConnect = function () {
   return Promise.resolve()
 }
 
+LocalStore.prototype.has = async function(k) {
+  return !!(await this.get(k))
```
Comment (Julien-laville) [inline src/libs/local_store.js]: that's golf

File: src/libs/local_store.js [hunk 8]
```diff
@@ -19,15 +19,31 @@ LocalStore.prototype.getAllPois = function() {
...
+  return Promise.resolve(null)
+}
+
+LocalStore.prototype.registerApp = function() {
   console.log('local storage doesn\'t support register method')
   return Promise.resolve()
 }
 
+LocalStore.prototype.unregisterApp = function() {
+  return Promise.resolve()
```
Comment (Julien-laville) [inline src/libs/local_store.js]: a think declare as async and return true / false is equivalent and more clear

File: src/libs/masq.js [hunk 9]
```diff
@@ -1 +1,112 @@
...
+
+  const filteredKeys = Object.keys(list).reduce((filtered, k) => {
+    if (Poi.isPoiCompliantKey(k)) {
+      filtered.push(k)
+    }
+    return filtered
+  }, [])
+
+  if (filteredKeys.length === 0) {
+    return Promise.resolve([])
```
Comment (Julien-laville) [inline src/libs/masq.js]: i think the async keyword is enought so you can return [] directly (can you confirm ?)

File: src/libs/masq.js [hunk 10]
```diff
@@ -1 +1,112 @@
...
+  }
+
+  const valuePromises = filteredKeys.map(k => {
```
Comment (Julien-laville) [inline src/libs/masq.js]: i dont understand what this const is

File: src/libs/masq.js [hunk 11]
```diff
@@ -1 +1,112 @@
...
+    return this.masq.get(k)
+  })
+
+  const values = await Promise.all(valuePromises).catch(
```
Comment (Julien-laville) [inline src/libs/masq.js]: i think we can use the try catch syntax here instead of mixin old Promise then & await syntax

File: src/libs/masq.js [hunk 12]
```diff
@@ -1 +1,112 @@
...
+}
+
+MasqStore.prototype.registerApp = async function(apps) {
+  // connect to Masq
+  window.open(this.loginLink)
+  await this.masq.logIntoMasq(false)
+}
+
+MasqStore.prototype.unregisterApp = async function() {
+  await this.masq.signout()
```
Comment (Julien-laville) [inline src/libs/masq.js]: i'm confuse with signout logout & unregister

File: src/libs/masq.js [hunk 13]
```diff
@@ -1 +1,112 @@
...
+}
+
+MasqStore.prototype.onConnect = async function () {
+  // executed when maps is opened
+  this.masq = new Masq(this.config.title, this.config.desc, this.config.icon)
+  if (this.masq.isLoggedIn()) {
+    await this.masq.connectToMasq()
```
Comment (Julien-laville) [inline src/libs/masq.js]: you can return here

File: src/libs/masq.js [hunk 14]
```diff
@@ -1 +1,112 @@
...
+  } else {
+    this.loginLink = await this.masq.getLoginLink()
```
Comment (Julien-laville) [inline src/libs/masq.js]: & here

File: src/libs/masq.js [hunk 15]
```diff
@@ -1 +1,112 @@
...
+  }
+
+  return
+}
+
+MasqStore.prototype.has = async function(k) {
+  return !!(await this.get(k))
```
Comment (Julien-laville) [inline src/libs/masq.js]: this is golf 🏌

File: src/main.js [hunk 16]
```diff
@@ -5,12 +5,16 @@ import './libs/actions'
...
 import UrlState from "./proxies/url_state";
+import Store from "./adapters/store";
 
 
 (async function main() {
   new I18n()
   await setLang()
 
+  const store = new Store()
+  await store.onConnect()
```
Comment (Julien-laville) [inline src/main.js]: are we sure we want to freeze app start-up to get store data ?

File: src/panel/register_masq.js [hunk 17]
```diff
@@ -8,19 +8,57 @@ function RegisterMasqPanel() {
   this.store = new Store()
   this.panel = new Panel(this, RegisterMasqPanelView)
   this.isActive = false
+  this.registering = false
+  this.username = null
+  this.profileImage = null
+
+  this.isRegistered = false
+  this.store.isRegistered().then(async (b) => {
```
Comment (Julien-laville) [inline src/panel/register_masq.js]: move to an async function

File: src/panel/register_masq.js [hunk 18]
```diff
@@ -8,19 +8,57 @@ function RegisterMasqPanel() {
...
+
+RegisterMasqPanel.prototype.unregister = async function () {
+  await this.store.unregister()
+  this.isRegistered = await this.store.isRegistered()
+  if (this.isRegistered) {
+    const { username, profileImage } = await this.store.getUserInfo()
+    this.username = username
+    this.profileImage = profileImage
+  }
+  window.location.reload()
```
Comment (Julien-laville) [inline src/panel/register_masq.js]: Oh waw the uge one

File: src/views/register_masq.dot [hunk 19]
```diff
@@ -1,11 +1,31 @@
...
-  <p>
-    {{= _('We use MASQ to keep your preference data safe', 'mask') }}
-  </p>
-  <br>
-  <button class="register_masq_panel__action" {{= click(this.register, this) }}>
-    {{= _('Register', 'mask') }}</button>
-</div>
\ No newline at end of file
+<div class="register_masq_panel">
+  {{? (!this.isRegistered)}}
```
Comment (Julien-laville) [inline src/views/register_masq.dot]: why (

File: src/views/register_masq.dot [hunk 20]
```diff
@@ -1,11 +1,31 @@
...
+    <h1 class="register_masq_panel__title">
+      {{= _('Register with masq', 'masq') }}
+    </h1>
+    <p>
+      {{= _('We use MASQ to keep your personal data safe', 'masq') }}
+    </p>
+    {{? (!this.registering)}}
```
Comment (Julien-laville) [inline src/views/register_masq.dot]: is ( necessary ?

File: src/adapters/store.js [hunk 21]
```diff
@@ -3,111 +3,169 @@ import Error from '../adapters/error'
...
 import ExtendedString from "../libs/string";
 
-let moduleConfig = nconf.get().store
+export default class Store {
 
-const AbStore = require(`../libs/${moduleConfig.name}`)
-const abstractStore = new AbStore(moduleConfig.endpoint)
+  constructor() {
+    // get store from window if already initialized
+    if (window.__store) {
```
Comment (Julien-laville) [inline src/adapters/store.js]: ok for me, but i need to explore a little this solution ( could be better than mine ;) )

File: src/adapters/store.js [hunk 22]
```diff
@@ -3,111 +3,169 @@ import Error from '../adapters/error'
...
-    })
+    // init stores
+    const LocalStore = require(`../libs/local_store`)
+    this.localStore = new LocalStore()
+    this.loggedIn = false
+    this.abstractStore = this.localStore
+    const MasqStore = require(`../libs/masq`)
+    this.masqConfig = nconf.get().masq
+    this.masqStore = new MasqStore(this.masqConfig)
+    const masqInitPromise = this.masqStore.onConnect()
```
Comment (Julien-laville) [inline src/adapters/store.js]: unused

File: src/adapters/store.js [hunk 23]
```diff
@@ -3,111 +3,169 @@ import Error from '../adapters/error'
...
+    const masqInitialized = false
```
Comment (Julien-laville) [inline src/adapters/store.js]: unused

File: src/adapters/store.js [hunk 24]
```diff
@@ -3,111 +3,169 @@ import Error from '../adapters/error'
...
-      this.del(poi)
-    })
-    listen('store_center', (loc) => {
-      this.setLastLocation(loc)
-    })
-    listen('store_clear', () => {
-      this.clear()
-    })
+    // init stores
+    const LocalStore = require(`../libs/local_store`)
```
Comment (Julien-laville) [inline src/adapters/store.js]: i'm not fond of dynamic require plus the syntax is the old one

File: src/adapters/store.js (hunk 2 above)
Comment (sujeebant) [reply src/adapters/store.js]: checkRegistered is not needed anymore with the new logic of calling localStorage until logged into masq

File: src/adapters/store.js (hunk 3 above)
Comment (sujeebant) [reply src/adapters/store.js]: :+1:

File: src/libs/local_store.js (hunk 8 above)
Comment (sujeebant) [reply src/libs/local_store.js]: :+1:

File: src/adapters/store.js [hunk 25]
```diff
@@ -3,111 +3,169 @@ import Error from '../adapters/error'
...
+        url: window.location.origin + window.location.pathname,
+        title: this.masqConfig.title,
+        desc: this.masqConfig.desc,
+        icon: this.masqConfig.icon
+      }
+
+      await this.masqStore.login(loginParams)
+    } catch (e) {
+      Error.sendOnce('store', 'login', 'error logging in', e)
+      throw e
```
Comment (Julien-laville) [inline src/adapters/store.js]: this error is catched twice (here and masqLoginPanel)

File: src/libs/masq.js (hunk 9 above)
Comment (sujeebant) [reply src/libs/masq.js]: Yes you're right !

File: src/libs/masq.js (hunk 10 above)
Comment (sujeebant) [reply src/libs/masq.js]: the array of promises that resolve to the values corresponding to the keys we are getting

File: src/adapters/store.js [hunk 26]
```diff
@@ -3,111 +3,169 @@ import Error from '../adapters/error'
...
-    title: moduleConfig.masq.title,
-    desc: moduleConfig.masq.desc,
-    icon: moduleConfig.masq.icon
+  async getAllPois() {
+    await this.checkInit()
+    try {
+      return await this.abstractStore.getAllPois()
+    } catch (e) {
+      Error.sendOnce('store', 'getAllPois', 'error getting pois', e)
+      throw e
```
Comment (Julien-laville) [inline src/adapters/store.js]: this file know wich store is used, so i prefer a more explicit error with the current store used, plus it's not necessary to re throw the excpetion, you can handle here with return []

File: src/libs/masq.js (hunk 11 above)
Comment (sujeebant) [reply src/libs/masq.js]: ok !

File: src/libs/masq.js (hunk 12 above)
Comment (sujeebant) [reply src/libs/masq.js]: Everything has been renamed to use login/logout

File: src/libs/local_store.js [hunk 27]
```diff
@@ -27,15 +27,31 @@ LocalStore.prototype.getAllPois = function() {
...
+LocalStore.prototype.isLoggedIn = function() {
+  return Promise.resolve(false)
+}
+
 LocalStore.prototype.onConnect = function () {
   return Promise.resolve()
 }
 
+LocalStore.prototype.has = async function(k) {
+  return !!(await this.get(k))
```
Comment (Julien-laville) [inline src/libs/local_store.js]: i prefer Boolean(

File: src/main.js (hunk 16 above)
Comment (sujeebant) [reply src/main.js]: changed to start execute onConnect at instantiation and await the resolution of the promise only when needed (at method execution)

File: src/libs/masq.js [hunk 28]
```diff
@@ -1 +1,112 @@
-module.exports = require('masq-client/dist/masq')
+const Masq = require('masq-lib')
+
+const Error = require('../adapters/error').default
```
Comment (Julien-laville) [inline src/libs/masq.js]: can you use the new import syntax when it's possible (or maybe it is not)

File: src/libs/masq.js (hunk 11 above)
Comment (Julien-laville) [inline src/libs/masq.js]: await catch then async mix here

File: src/libs/masq.js [hunk 29]
```diff
@@ -1 +1,112 @@
...
+    return this.masq.get(k)
+  })
+
+  const values = await Promise.all(valuePromises).catch(
+    (e) => {
+      handleError('getAllPois', 'error getting pois', e)
+      throw e
+    })
+
+  return values
```
Comment (Julien-laville) [inline src/libs/masq.js]: value is redundant

File: src/libs/masq.js [hunk 30]
```diff
@@ -1 +1,112 @@
...
+  return !!(await this.get(k))
+}
+
+MasqStore.prototype.get = async function(k) {
+  const value = await this.masq.get(k).catch(
+    (e) => {
+      handleError('get', `error parsing item with key ${k}`, e)
+      throw e
+    })
+  return value
```
Comment (Julien-laville) [inline src/libs/masq.js]: value is redundant

File: src/libs/masq.js [hunk 31]
```diff
@@ -1 +1,112 @@
...
+  return Promise.resolve()
+}
+
+MasqStore.prototype.del = async function(k) {
+  await this.masq.del(k).catch(
+    (e) => {
+      handleError('del', 'error removing item', e)
+      throw e
+    })
+  return
```
Comment (Julien-laville) [inline src/libs/masq.js]: si return pertinent ?

File: src/libs/masq.js [hunk 32]
```diff
@@ -1 +1,112 @@
...
+  return value
+}
+
+MasqStore.prototype.set = async function(k, v) {
+  await this.masq.put(k, v).catch(
+    (e) => {
+      handleError('set', 'error setting item', e)
+      throw e
+    })
+  return
```
Comment (Julien-laville) [inline src/libs/masq.js]: si return pertinent ?

File: src/libs/panel.js [hunk 33]
```diff
@@ -111,4 +111,4 @@ Panel.prototype.animate = async function(t, selector, style = {}) {
   })
 }
 
-export default Panel
\ No newline at end of file
+export default Panel
```
Comment (Julien-laville) [inline src/libs/panel.js]: thank's :3

File: src/panel/login_masq.js [hunk 34]
```diff
@@ -0,0 +1,66 @@
...
+export default class LoginMasqPanel {
+  constructor() {
+    this.panel = new Panel(this, LoginMasqPanelView)
+    if(moduleConfig.name === 'masq') {
+      this.store = new Store()
+      this.isActive = false
+      this.loggingIn = false
+      this.username = null
+      this.profileImage = null
+      listen('login_panel__show', () => {
```
Comment (Julien-laville) [inline src/panel/login_masq.js]: you can remove _panel

File: src/panel/login_masq.js [hunk 35]
```diff
@@ -0,0 +1,66 @@
...
+        this.panel.animate(.25, '.login_masq_panel', {top : '100px'})
+      })
+
+      this.isLoggedIn = false
+      this.store.isLoggedIn().then(async (b) => {
```
Comment (Julien-laville) [inline src/panel/login_masq.js]: can you move this to an async function so you can clean async syntax

File: src/panel/login_masq.js [hunk 36]
```diff
@@ -0,0 +1,66 @@
...
+    try {
+      await this.store.login()
+      this.isLoggedIn = await this.store.isLoggedIn()
+      if (this.isLoggedIn) {
+        const { username, profileImage } = await this.store.getUserInfo()
+        this.username = username
+        this.profileImage = profileImage
+      }
+    } catch(e) {
+      Error.sendOnce('login_masq', 'login', 'error loggingIn masq', e)
```
Comment (Julien-laville) [inline src/panel/login_masq.js]: i think this error is already handle in store 
i prefer avoid sending 2 messages with the same error

File: src/scss/includes/panels/login.scss [hunk 37]
```diff
@@ -0,0 +1,41 @@
+.login_masq_panel {
```
Comment (Julien-laville) [inline src/scss/includes/panels/login.scss]: very nice css !

File: src/adapters/store.js (hunk 21 above)
Comment (Julien-laville) [reply src/adapters/store.js]: after a little of reading, i'm completely ok with the approach, it's fully correct !
but 
the singleton can be recognized by it's signature (getInstance) i let you decide which one you prefer
by my side i have learnt something

File: src/adapters/store.js (hunk 22 above)
Comment (sujeebant) [reply src/adapters/store.js]: It was supposed to be an attribute...

File: src/adapters/store.js (hunk 23 above)
Comment (sujeebant) [reply src/adapters/store.js]: It's an error on my part, it was supposed to be an attribute used in checkInit

File: src/adapters/store.js (hunk 24 above)
Comment (sujeebant) [reply src/adapters/store.js]: :+1:

File: src/libs/masq.js (hunk 28 above)
Comment (sujeebant) [reply src/libs/masq.js]: moved the whole file to es6 syntax !

File: src/libs/local_store.js (hunk 27 above)
Comment (sujeebant) [reply src/libs/local_store.js]: done in local_store.js and masq.js

File: src/libs/masq.js (hunk 11 above)
Comment (sujeebant) [reply src/libs/masq.js]: fixed !

File: src/libs/masq.js (hunk 29 above)
Comment (sujeebant) [reply src/libs/masq.js]: :+1:

File: src/libs/masq.js (hunk 30 above)
Comment (sujeebant) [reply src/libs/masq.js]: :+1:

File: src/libs/masq.js (hunk 32 above)
Comment (sujeebant) [reply src/libs/masq.js]: cleaned unnecessary return statements in masq.js

File: src/libs/local_store.js (hunk 7 above)
Comment (sujeebant) [reply src/libs/local_store.js]: cleaned !

File: src/libs/masq.js (hunk 15 above)
Comment (sujeebant) [reply src/libs/masq.js]: cleaned !

File: src/libs/masq.js (hunk 31 above)
Comment (sujeebant) [reply src/libs/masq.js]: cleaned unnecessary return statements in masq.js

File: src/panel/login_masq.js (hunk 35 above)
Comment (sujeebant) [reply src/panel/login_masq.js]: moved to an async function called in the constructor without await

File: src/scss/includes/panels/login.scss (hunk 37 above)
Comment (sujeebant) [reply src/scss/includes/panels/login.scss]: just renamed some class names

File: src/views/register_masq.dot (hunk 19 above)
Comment (sujeebant) [reply src/views/register_masq.dot]: You are right '(' is not necessary

File: src/views/register_masq.dot (hunk 20 above)
Comment (sujeebant) [reply src/views/register_masq.dot]: You are right '(' is not necessary

File: src/adapters/store.js (hunk 21 above)
Comment (sujeebant) [reply src/adapters/store.js]: As you have pointed out this implementation hides the fact that Store is a singleton, but the user of the class is able to just call 'new Store()' to get the Store instance.
I guess the two approaches are valid, I will leave it as it is, we can change it later if needed.

File: src/panel/register_masq.js (hunk 18 above)
Comment (sujeebant) [reply src/panel/register_masq.js]: :man_facepalming: I wanted to refresh the view and favorite list...
Changed to fire an event on login and logout caught by Favorite panel to refresh the list

File: package.json [hunk 38]
```diff
@@ -89,6 +88,7 @@
     "ejs": "^2.6.1",
     "express": "^4.16.3",
     "express-static-gzip": "^1.1.3",
+    "masq-lib": "git+https://github.com/QwantResearch/masq-lib.git#maps-integration",
```
Comment (amatissart) [inline package.json]: I think this dependency is only required on build, and should be moved to "devDependencies".  

This currently blocks the docker image build, as it also depends on other devDependencies.

Comment (sujeebant) [reply package.json]: You're righ ! I moved masq-lib to devDependencies

File: src/adapters/scene.js [hunk 39]
```diff
@@ -123,7 +123,7 @@ Scene.prototype.initMapBox = function () {
       let lng = this.mb.getCenter().lng
```
[the remark this answers is not in the document]
Comment (sujeebant) [reply src/adapters/scene.js]: done

File: src/panel/login_masq.js [hunk 40]
```diff
@@ -0,0 +1,64 @@
+import LoginMasqPanelView from '../views/login_masq.dot'
+import Panel from "../libs/panel";
+import Store from "../adapters/store"
+import Error from '../adapters/error'
+import nconf from "../../local_modules/nconf_getter";
+let moduleConfig = nconf.get().store
```
[the remark this answers is not in the document]
Comment (sujeebant) [reply src/panel/login_masq.js]: removed nconf import and moduleConfig declaration

File: src/panel/login_masq.js (hunk 34 above)
Comment (sujeebant) [reply src/panel/login_masq.js]: removed the listener

File: src/adapters/store.js (hunk 25 above)
Comment (sujeebant) [reply src/adapters/store.js]: removed error logging in panel/login_masq

File: src/panel/login_masq.js (hunk 36 above)
Comment (sujeebant) [reply src/panel/login_masq.js]: okay ! removed the second error logging in panel/login_masq as the first one can be more specific with the name of the store which is used

File: src/adapters/store.js [hunk 41]
```diff
@@ -2,112 +2,170 @@ import nconf from '@qwant/nconf-getter'
...
+    this.loggedIn = false
+    this.abstractStoreStr = 'local_store'
+    this.abstractStore = this.localStore
+    this.masqConfig = nconf.get().masq
+    if (this.masqConfig.enabled) {
+      this.masqStore = new MasqStore(this.masqConfig)
+      if (this.masqStore.isLoggedIn()) {
+        this.abstractStore = this.masqStore
+      }
+    }
```
Comment (Julien-laville) [inline src/adapters/store.js]: else new localStore 
should be more acceptable, here you instanciate local store, and you instanciate masqStore

File: src/adapters/store.js [hunk 42]
```diff
@@ -2,112 +2,170 @@ import nconf from '@qwant/nconf-getter'
...
-    listen('store_center', (loc) => {
-      this.setLastLocation(loc)
-    })
-    listen('store_clear', () => {
-      this.clear()
-    })
+    // init stores
+    this.localStore = new LocalStore()
+    this.loggedIn = false
+    this.abstractStoreStr = 'local_store'
```
Comment (Julien-laville) [inline src/adapters/store.js]: why not displace the logic to currentStore.getName or something

File: src/adapters/store.js [hunk 43]
```diff
@@ -2,112 +2,170 @@ import nconf from '@qwant/nconf-getter'
...
+    this.abstractStore = this.localStore
+    this.masqConfig = nconf.get().masq
+    if (this.masqConfig.enabled) {
+      this.masqStore = new MasqStore(this.masqConfig)
```
Comment (Julien-laville) [inline src/adapters/store.js]: you should let masqStore handle is own config internaly

File: src/adapters/store.js [hunk 44]
```diff
@@ -2,112 +2,170 @@ import nconf from '@qwant/nconf-getter'
...
+import MasqStore from "../libs/masq"
 
-let moduleConfig = nconf.get().store
+export default class Store {
 
-const AbStore = require(`../libs/${moduleConfig.name}`)
-const abstractStore = new AbStore(moduleConfig.endpoint)
+  constructor() {
+    // get store from window if already initialized
+    if (window.__store) {
```
Comment (Julien-laville) [inline src/adapters/store.js]: I have gather more data about our "problem" 
https://stackoverflow.com/questions/519520/difference-between-static-class-and-singleton-pattern
the fact that this class is never used as instance made me think that we should declare it as a "static class"
now it's a little late

**Pull request merged**